### PR TITLE
fix: error message when response is not 201

### PR DESCRIPTION
**Pull request by d2lam**
Seeing `error: failed to post build event for job 120068: null`

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -207,11 +207,15 @@ class ExecutorQueue extends Executor {
         }
 
         return new Promise((resolve, reject) => {
-            req(options, (err, response) => {
+            req(options, (err, response, body) => {
```
Comment (jithine) [inline index.js:210]: what's in err ?

File: index.js [hunk 2]
```diff
@@ -207,11 +207,15 @@ class ExecutorQueue extends Executor {
...
                 if (!err && response.statusCode === 201) {
                     return resolve(response);
                 }
 
+                if (response.statusCode !== 201) {
```
Comment (jithine) [inline index.js]: if we enter this condition, the following `reject` never gets executed/

Comment (d2lam) [reply index.js]: why?

**Review by jithine — APPROVED** [review]

File: index.js (hunk 1 above)
Comment (d2lam) [reply index.js:210]: `err` is only when something is wrong with request module.
most response, the err is in the body like this:
```
Error : null
Body : {"statusCode":404,"error":"Not Found","message":"Not Found"}
```

**Pull request merged**